src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -659,12 +644,14 @@ def engine_forward(
             finished_reason = []
             streaming = context.get("streaming")
             generation_config = context.get("generation_config")
+            deterministic = not generation_config.do_sample
```

Comment:
Will generation config always be present in context? also could we raise an exception if generation_config is not an instance of GenerationConfig? or is that check already done somewhere else?

Reply:
generation config will always be present, because if nothing, we'll provide our defaults and all the checks are done as part of the processing